src/sentry/api/endpoints/organization_details.py

```diff
@@ -380,6 +387,13 @@ def save(self):
             org.name = data["name"]
         if "slug" in data:
             org.slug = data["slug"]
+        if "provider" in data and "providerConfig" in data:
+            provider_name = data["provider"]
```

Comment:
Is `provider` the correct right name for this key?

Reply:
What's `provider_name`? Is it just any string? Should we validate anything in it?

Reply:
We have a list of valid `provider_name`, perhaps we should enforce this in this public-facing API? https://github.com/getsentry/sentry/blob/6e2f44a330d63de7b59e913cd3c78f9c1abb20ac/src/sentry/auth/providers/saml2/generic/provider.py#L7

Reply:
Changed `provider` -> `authProvider`

Also added authProvider validator that checks if provider's name is one of `("github", "google", "saml2", "jumpcloud", "okta", "onelogin", "rippling")`